src/datasources/exchange-api/exchange-api.service.ts

```diff
@@ -25,6 +27,14 @@ export class ExchangeApi {
   ) {
     this.baseUrl = this.configurationService.get<string>('exchange.baseUri');
     this.apiKey = this.configurationService.get<string>('exchange.apiKey');
+
```

Comment:
This is not strictly necessary, but TS compiler was complaining about `baseUrl`/`apiKey` could be `undefined`, so I realised it could be nice to warn about this situation at startup time if it happens. I chose to use `logger.warn` but it could be `logger.error` as well, no strong preference here.

Reply:
I would suggest the following: setting both as `getOrThrow`. This class should only work if we have at least these set.

If you use that you might be able to remove the `undefined` from the `baseUrl` and `apiKey`

Reply:
I agree, much better solution, thanks!